### PR TITLE
Create custom methods in model classes

### PR DESCRIPTION
**Pull request by yasinabdmahmood**
### Created models
-  Created model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
-  Set up associations between models.

### Used models to insert data
- Created two users by running the following code:
    ```ruby
       first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
       second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
    ```
- Created  4 posts written by the first  user  by running the following code:
    ```ruby
       first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
    ```
- Create  6 posts comments for the first  post by running the following code:
    ```ruby
       Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
    ```

### Created custom methods
the models include some additional methods.
-  Users
    - A method that returns the 3 most recent posts for a given user.
-  Posts
    - A method that updates the posts counter for a user.
    - A method which returns the 5 most recent comments for a given post.
-  Comments
    - A method that updates the comments counter for a post.
-  Likes
    - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,2 +1,8 @@
 class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
```
Comment (mukhammadrizooff) [inline app/models/comment.rb]: - [ ] [REQUIRED] Good job so far! The `associations` between `models` should be set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Please implement my suggestions below 🙏 

```ruby
# app/models/comment.rb
belongs_to :author, class_name: 'User'
belongs_to :post
```

File: app/models/like.rb [hunk 2]
```diff
@@ -1,2 +1,8 @@
 class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
```
Comment (mukhammadrizooff) [inline app/models/like.rb]: - [ ] [REQUIRED] Good job so far! The `associations` between `models` should be set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Please implement my suggestions below 🙏 

```ruby
# app/models/like.rb
belongs_to :author, class_name: 'User'
belongs_to :post
```

File: app/models/post.rb [hunk 3]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :user
+  has_many :likes
+  has_many :comments
```
Comment (mukhammadrizooff) [inline app/models/post.rb]: - [ ] [REQUIRED] Good job so far! The `associations` between `models` should be set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Please implement my suggestions below 🙏 

```ruby
# app/models/post.rb
belongs_to :author, class_name: 'User'
has_many :comments
has_many :likes
```

File: app/models/user.rb [hunk 4]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  has_many :posts
```
Comment (mukhammadrizooff) [inline app/models/user.rb]: - [ ] [REQUIRED] Good job so far! The `associations` between `models` should be set up correctly according to the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png). Please implement my suggestions below 🙏 

```ruby
# app/models/user.rb
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @yasinabdmahmood, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @yasinabdmahmood 



Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.
- [ ] [OPTIONAL]⚠️ you have create associations as required however you still need to change your references to author in the migration file as well in order to have a foreign_key called  `author_id ` in the association's tables like: ` add_reference: :(table name) :author, foreign_key: { to_table: 'users' }`
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**